packages/angular/src/migrations/update-13-3-0/update-mfe-webpack-config.ts

```diff
@@ -0,0 +1,128 @@
+import { Tree } from '@nrwl/devkit';
+import { logger, formatFiles } from '@nrwl/devkit';
+import { tsquery } from '@phenomnomnominal/tsquery';
+import { forEachExecutorOptions } from '@nrwl/workspace/src/utilities/executor-options-utils';
+import { Identifier, Node, StringLiteral } from 'typescript';
+
+export default async function (tree: Tree) {
+  const NRWL_WEBPACK_BROWSER_BUILDER = '@nrwl/angular:webpack-browser';
+  const CUSTOM_WEBPACK_OPTION = 'customWebpackConfig';
+
+  forEachExecutorOptions(
+    tree,
+    NRWL_WEBPACK_BROWSER_BUILDER,
+    (opts, projectName) => {
+      // Update the webpack config
+      const webpackPath = opts[CUSTOM_WEBPACK_OPTION].path;
+      if (!tree.exists(webpackPath)) {
+        logger.warn(
+          `Webpack config file for project: ${projectName} does not exist. Skipping project.`
+        );
+        return;
+      }
+      const webpackConfig = tree.read(webpackPath, 'utf-8');
+      if (!webpackConfig.includes('ModuleFederationPlugin')) {
+        logger.warn(
+          `Webpack config file for project: ${projectName} is not using Module Federation. Skipping project.`
+        );
+        return;
+      }
+
+      let updatedWebpackFile = addExperimentsObject(webpackConfig);
+      updatedWebpackFile = addLibraryModuleType(updatedWebpackFile);
+      updatedWebpackFile = removeRemoteName(updatedWebpackFile);
+      tree.write(webpackPath, updatedWebpackFile);
+    }
+  );
+
+  await formatFiles(tree);
+}
+
+export function addExperimentsObject(webpackConfig: string) {
+  const WEBPACK_EXPERIMENTS_QUERY =
+    'PropertyAssignment > Identifier[name=experiments] ~ ObjectLiteralExpression';
+  const ast = tsquery.ast(webpackConfig);
+  const webpackExperimentsNode = tsquery(ast, WEBPACK_EXPERIMENTS_QUERY);
+
+  if (webpackExperimentsNode && webpackExperimentsNode.length > 0) {
```

Comment:
Shouldn't we still check if the specific `outputModule` experiment is configured before bailing out?
The `experiments` might have other experiments inside and we should still add `outputModule` if it's not one of them, isn't it?

Reply:
Good point!